src/main/java/org/broadinstitute/hellbender/engine/FeatureContext.java

```diff
@@ -107,11 +110,77 @@ public SimpleInterval getInterval() {
+     *         this FeatureContext's query interval. Empty List if there is no backing data source and/or interval.
+     */
+    public <T extends Feature> List<T> getValues( final FeatureInput<T> featureDescriptor, final int windowLeadingBases, final int windowTrailingBases) {
+        return getValues(featureDescriptor, getWindow(windowLeadingBases, windowTrailingBases));
+    }
+
+    private <T extends Feature> List<T> getValues( final FeatureInput<T> featureDescriptor, final SimpleInterval window) {
+        if ( featureManager == null || window == null || featureDescriptor == null) {
             return Collections.<T>emptyList();
         }
+        return featureManager.getFeatures(featureDescriptor, window);
+    }
 
-        return featureManager.getFeatures(featureDescriptor, interval);
+    /**
+     * Gets our query interval (the interval that all Features returned by this FeatureContext overlap).
+     * Null if this context has no interval.
+     *
+     * @param windowLeadingBases Number of extra bases to include before the start of our interval. Must be >= 0.
+     * @param windowTrailingBases Number of extra bases to include after the end of our interval. Must be >= 0.
+     * @return full expanded window of bases spanned by this context as a SimpleInterval
+     *         (will be null if this context has no interval)
+     */
+    public SimpleInterval getWindow(final int windowLeadingBases, final int windowTrailingBases){
+        if( windowLeadingBases < 0 ) throw new GATKException("Window starts after the current interval");
+        if( windowTrailingBases < 0 ) throw new GATKException("Window ends before the current interval");
+
+        if (interval == null) {
+            return null;
+        }
+        if (windowLeadingBases == 0 && windowTrailingBases == 0){
+            return interval;
+        }
+        return new SimpleInterval(interval.getContig(), windowStart(interval, windowLeadingBases), windowStop(interval, windowTrailingBases));
+    }
+
+    /**
+     * Determines the start of the expanded reference window, bounded by 1.
+     *
+     * @param locus The locus to expand.
+     * @param windowLeadingBases number of bases to attempt to expand relative to the locus start (>= 0)
+     * @return The start of the expanded window.
+     */
+    private int windowStart( final SimpleInterval locus, final int windowLeadingBases ) {
+        return Math.max(locus.getStart() - windowLeadingBases, 1);
+    }
+
+    /**
+     * Determines the stop of the expanded reference window.
+     *
+     * @param locus The locus to expand.
+     * @param windowTrailingBases number of bases to attempt to expand relative to the locus end (>= 0)
+     * @return The end of the expanded window.
+     */
+    private int windowStop( final SimpleInterval locus, final int windowTrailingBases ) {
+        //XXX what about contig end? Is this a problem?
+        return locus.getEnd() + windowTrailingBases;
```

Comment:
Yes, it's potentially a problem, though I believe that the htsjdk `FeatureReader`, at least, can handle queries off the ends of contigs without blowing up or returning incorrect results. Can you add a unit test to confirm that?

Note that the `ReferenceContext` version of these methods crops the window stop at contig boundaries, but here in `FeatureContext` we may not have a sequence dictionary at all.